relayer-cli/src/utils/relay.ts

```
@@ -35,101 +39,113 @@ const getCount = async (veaOutbox: VeaOutboxArbToEth | VeaOutboxArbToGnosis, cha
-    routeParams.veaOutboxAddress,
-    process.env.PRIVATE_KEY,
-    routeParams.rpcOutbox,
-    chainId
-  );
+  const bridgeConfig = fetchBridgeConfig(chainId);
+  const privateKey = process.env.PRIVATE_KEY;
+  const { batcherAddress, veaContracts, rpcOutbox } = bridgeConfig;
+  const veaInboxAddress = veaContracts[network].veaInbox.address;
+  const veaOutboxAddress = veaContracts[network].veaOutbox.address;
+
+  const batcher = fetchBatcher(batcherAddress, privateKey, rpcOutbox);
+  const veaOutbox = fetchVeaOutbox(veaOutboxAddress, privateKey, rpcOutbox, chainId, network);
   const count = await fetchCount(veaOutbox, chainId);
 
   while (nonce < count) {
     let batchMessages = 0;
-    let txns = [];
+    let targets: string[] = [];
+    let values: number[] = [];
+    let datas: string[] = [];
+
     while (batchMessages < maxBatchSize && nonce < count) {
       const isMsgRelayed = await veaOutbox.isMsgRelayed(nonce);
       if (isMsgRelayed) {
         nonce++;
         continue;
       }
-      const [proof, [to, data]] = await Promise.all([
-        fetchProofAtCount(chainId, nonce, count),
-        fetchMessageDataToRelay(chainId, routeParams.veaInboxAddress, nonce),
+      const [proof, messageData] = await Promise.all([
+        fetchProofAtCount(chainId, nonce, count, veaInboxAddress),
+        fetchMessageDataToRelay(chainId, veaInboxAddress, nonce),
       ]);
+      const [to, data] = messageData;
       try {
-        await veaOutboxInstance.methods.sendMessage(proof, nonce, to, data).call();
+        await veaOutbox.sendMessage.staticCall(proof, nonce, to, data);
+        const callData = veaOutbox.interface.encodeFunctionData("sendMessage", [proof, nonce, to, data]);
+        datas.push(callData);
+        targets.push(veaOutboxAddress);
+        values.push(0);
+        batchMessages += 1;
+        nonce++;
       } catch {
+        emitter.emit(BotEvents.MESSAGE_EXECUTION_FAILED, nonce);
         nonce++;
-        continue;
```

I am not sure what happens when a `MESSAGE_EXECUTION_FAILED` event is emitted, but it will be for sure a better alternative that just swallowing the errors in a silent `continue`. Nice refactor here.

Any reason not to capture the error and provide that context to the emitted event? We may benefit from seeing a stack trace and/or error message to figure out what went wrong.

---

Ideally we dont want a batch to fail for a single failing msg execution which depends on the destination contract of the message. I added this event in case we want to monitor failing messages and add a retry mechanism for them after some time. I dont think the error message would be relevant here as it would always be a result of failed execution on the destination contract OR when the bot runs out of gas.